site/en/responsible/docs/safeguards/shieldgemma2_on_huggingface.ipynb

```diff
@@ -0,0 +1,214 @@
+        "\n",
+        "* **Dangerous Content**:  The image shall not contain content that facilitates or encourages activities that could cause real-world harm (e.g., building firearms and explosive devices, promotion of terrorism, instructions for suicide).\n",
+        "\n",
+        "* **Sexually Explicit**: The image shall not contain content that depicts explicit or graphic sexual acts (e.g., pornography, erotic nudity, depictions of rape or sexual assault).\n",
+        "\n",
+        "* **Violence/Gore**: The image shall not contain content that depicts shocking, sensational, or gratuitous violence (e.g., excessive blood and gore, gratuitous violence against animals, extreme injury or moment of death).\n",
+        "\n",
+        "This serves as a foundation, but users can provide customized safety policies as input to the model, allowing for fine-grained control and specific use-case requirements."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "t3aq-ToeAmRM"
+      },
+      "source": [
+        "# Supported Use Case\n",
+        "\n",
+        "**We recommend using `ShieldGemma 2` as an input filter to vision language models or as an output filter of image generation systems or both.**  ShieldGemma 2 offers the following key advantages:\n",
+        "\n",
+        "* **Policy-Aware Classification**: ShieldGemma 2 accepts both a user-defined safety policy and an image as input, providing classifications for both real and generated images, tailored to the specific policy guidelines.\n",
+        "* **Probability-Based Output and Thresholding**: ShieldGemma 2 outputs a probability score for its predictions, allowing downstream users to flexibly tune the classification threshold based on their specific use cases and risk tolerance. This enables a more nuanced and adaptable approach to safety classification.\n",
+        "\n",
+        "The input/output format are as follows:\n",
+        "* **Input**: Image + Prompt Instruction with policy definition\n",
+        "* **Output**: Probability of 'Yes'/'No' tokens, 'Yes' meaning that the image violated the specific policy. The higher the score, the higher the model's confidence that the image violates the specified policy."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0WhRozADVJos"
+      },
+      "source": [
+        "# Usage example"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "K_XERopLUZhk"
+      },
+      "outputs": [],
+      "source": [
+        "! pip install -q 'transformers>=4.50.0'"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Qg-Hy0ffbwvE"
+      },
+      "outputs": [],
+      "source": [
+        "! huggingface-cli login"
```

Comment:
This command takes input from stdin, which is fine for an interactive notebook but if this is published from github -> devsite, then it will try and run the notebook, and fail (presumably through timeout).

One way to avoid this is to save some output in the notebook. They're only run if no output is saved.

If this doesn't rely on being published to devsite automatically then this is fine too. Typically the docs in this repo do though.

Reply:
Thanks for the heads up. Commented out the code in this cell and left detailed `@markdown` instructions for the recommended ways to authenticate with HF Hub with a note to un-comment and run this cell as a fallback.

Reply:
Commenting out that cell will stop that cell from being executed, but if that cell is required to make the following cells work, then the notebook will still fail and won't get imported to devsite.

```
OSError: You are trying to access a gated repo.
Make sure to have access to it at https://huggingface.co/google/shieldgemma-2-4b-it.
401 Client Error. (Request ID: Root=1-67e1f1ed-2dd02b30690f1f2c40dc70ff;e41055b2-8a5f-4b73-9089-f43ec4a0dc64)
```

Notebooks are only run if there is no output saved, so save some output and publication will work.